### ISSUE TITLE
OAuth (Codeberg) does not allow secondary e-mails to be used

### ISSUE THREAD

**Issue by lucasmz-dev** [WeblateOrg/weblate#12167]
### Describe the issue

When using Codeberg's Weblate which only uses their own authentication with Codeberg, I am not able to use any secondary e-mails set up in the account.
I usually have an email used for my Git identity (git@lucasmz.dev), I usually set my commit e-mail on Hosted Weblate to it, unfortunately due to this issue, I am not able to do that on this type of instance, because only the main address is considered.

Switching the main e-mail on Codeberg to that one, and re-logging in their instance, makes it so that it is available, undoing it however, removes it again. 

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar filed issues in this repository.

### Steps to reproduce the behavior

1. Add a secondary email address to Codeberg
2. Log in to Weblate
3. Go to account settings
4. No email found

### Expected behavior

All email addresses associated with that account to be usable in Weblate.

### Screenshots

![image](https://github.com/user-attachments/assets/e1510987-bf3b-49fc-83d7-024e5ac42814)
![image](https://github.com/user-attachments/assets/64bf87ec-6239-433e-bbd9-41ff84257d98)

### How do you run Weblate?

Other

### Additional context

This is not an issue with hosted.weblate.org and GitHub for instance, if I were to use that.
Codeberg Translate does not allow the addition of other e-mail addresses into weblate directly.

**Comment by nijel** [comment]
Weblate has GitHub integration to fetch additional email addresses, something similar needs to be implemented here as well.

**Comment by Het4304** [comment]
I am interested in solving the issue

**Comment by nijel** [comment]
There are two possible approaches, depending on how Codeberg (Gitea API) exposes this:

1. When exposed in `/api/v1/user`, it can be implemented in python-social-auth here: https://github.com/python-social-auth/social-core/blob/d9554fa40e751c85ae60231fe2f5bd5a528c4452/social_core/backends/gitea.py#L36-L51
2. When there is a separate API call needed, it should be implemented in Weblate like we do for GitHub:
 
  https://github.com/WeblateOrg/weblate/blob/9a01e7d92c7bd1b37aedce381a072f3c003e3881/weblate/accounts/pipeline.py#L55-L87

**Comment by jlucas10** [comment]
Hi there! I'm a computer science student currently in the CodePath AI301 course, and I'd love to take this issue on as my first open-source contribution.

I saw the technical pointers provided regarding the OAuth pipeline in weblate/accounts/pipeline.py. I'd like to implement the Codeberg/Gitea secondary email fetching logic by mirroring the existing GitHub pipeline function.

Is this issue open for a PR? If so, I can get started on it this week!

**Comment by nijel** [comment]
Yes, pull requests are welcome. Meanwhile, the GitHub implementation was moved upstream (https://github.com/python-social-auth/social-core/pull/1710 and https://github.com/WeblateOrg/weblate/pull/19293) so it might be more useful to follow the same approach here.
